ql/src/java/org/apache/hadoop/hive/ql/Driver.java

```diff
@@ -2186,6 +2195,43 @@ private TaskRunner launchTask(Task<? extends Serializable> tsk, String queryId,
     return tskRun;
   }
 
+  private void enforceQueue(HiveConf conf) {
+    String queueEnforcerClassName = conf.getVar(HiveConf.ConfVars.QUEUE_ENFORCER_CLASS);
+    QueueEnforcer queueEnforcer = null;
+    try {
+      queueEnforcer = (QueueEnforcer) Class.forName(queueEnforcerClassName)
+          .newInstance();
+    } catch (Throwable e) {
+      throw new RuntimeException("Couldn't enforce queue due to " + e.getMessage());
```

Comment:
CLI main thread would exit if we throw runtime exception. Can we throw SQLException instead wherever we are throwing runtime exception?

Reply:
Don't we want CLI main thread to exit in such cases?